cfme/infrastructure/pxe.py

```diff
@@ -742,6 +782,26 @@ def create(self, name, provision_type, cancel=False):
         main_view.flash.assert_success_message(msg)
         return system_image_type
 
+    def delete(self, cancel=False, *sys_objs):
+        """
+        This methods deletes the System Image Type using select option,
+         hence can be used for multiple delete.
+
+        Args:
+            cancel: This is the boolean argument required for handle_alert
+            sys_objs: It's System Image Types object
+        """
+        view = navigate_to(self, 'All')
+        for sys_obj in sys_objs:
+            view.entities.row(Name=sys_obj.name)[0].click()
+        view.toolbar.configuration.item_select("Remove System Image Types", handle_alert=cancel)
```

Comment:
**Required**

You've got two calls to configuration.item_select, one before checking cancel and one after.

The call to item_select can be made before checking cancel, as long as `handle_alert` is passed correctly.

I think you should call `flash.assert_no_error` regardless of whether the delete was cancelled.

Reply:
made this change

Reply:
Close - need to pass `not cancel` to `handle_alert` for item_select.

Check out the docstring for handle_alert:
"handle_alert: How to handle alerts. None - no handling, True - confirm, False - dismiss."